playbooks/common/openshift-cluster/redeploy-certificates/registry.yml

```diff
@@ -46,12 +46,16 @@
 
   # Replace dc/docker-registry certificate secret contents if set.
   - block:
+    - name: Load lib_openshift modules
+      include_role:
+      name: lib_openshift
+
     - name: Retrieve registry service IP
-      command: >
-        {{ openshift.common.client_binary }} get service docker-registry
-        -o jsonpath='{.spec.clusterIP}'
-        --config={{ mktemp.stdout }}/admin.kubeconfig
-        -n default
+      oc_service:
+        namespace: default
+        name: docker-registry
+        state: list
+        kubeconfig: "{{ mktemp.stdout }}/admin.kubeconfig"
```

Comment:
The default value for the ``kubeconfig`` if ``/etc/origin/master/admin.kubeconfig``. Given that, we should add additional check at the beginning of the all play making sure ``{{ openshift.common.config_base }}`` equals ``/etc/origin``. Otherwise the module will most likely fail. Unless a master node has both ``/etc/origin/master/admin.kubeconfig`` and ``/SOMEPATH/origin/master/admin.kubeconfig`` and we want the latter case. In which case it will be very hard to debug.